### PR TITLE
Refactor cmd

### PR DESCRIPTION
**Pull request by garfield009**
## Description
refactor cmd modules and common path

## Related Issue
#175 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: client/dfget/dfget.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		end   time.Time
+	)
+	down, err := client.Download(ctx, request)
+	if err != nil {
+		return err
+	}
+	var (
+		result *dfdaemongrpc.DownResult
+	)
+	// todo showBar为true才使用
```
Comment (gaius-qi) [inline client/dfget/dfget.go:91]: Use English

File: cmd/cdn/cmd/root.go [hunk 2]
```diff
@@ -69,18 +64,17 @@ func Execute() {
...
 }
 
 func runCdnSystem() error {
 	// cdn system config values
 	s, _ := yaml.Marshal(cfg)
 	logger.Infof("cdn system configuration:\n%s", string(s))
 
-	// initialize verbose mode
-	common.InitVerboseMode(cfg.Verbose, cfg.PProfPort)
+	ff := common.InitMonitor(cfg.Verbose, cfg.PProfPort, cfg.Jaeger)
```
Comment (gaius-qi) [inline cmd/cdn/cmd/root.go:76]: What is `ff`?

File: cmd/common/common.go [hunk 3]
```diff
@@ -17,88 +17,186 @@
...
-	
-	if err := viper.BindPFlags(flagSet); err != nil {
-		panic(errors.Wrap(err, "bind flags to viper"))
+// config is a pointer to configuration struct.
+func InitCobra(cmd *cobra.Command, useConfigFile bool, config interface{}) {
+	rootName := cmd.Root().Name()
+	cobra.OnInitialize(func() { initConfig(useConfigFile, rootName, config) })
+
+	// Bind common flags,execute only once
+	if !isCommonFlagsBind {
```
Comment (gaius-qi) [inline cmd/common/common.go:69]: Why use `isCommonFlagsBind`?

File: cmd/common/common.go [hunk 4]
```diff
@@ -17,88 +17,186 @@
...
+		isCommonFlagsBind = true
+		rootFlags := cmd.Root().PersistentFlags()
+		// Add flags
+		rootFlags.Bool("console", false, "whether logger output records to the stdout")
+		rootFlags.Bool("verbose", false, "whether logger use debug level")
+		rootFlags.Int("pprof-port", -1, "listen port for pprof, 0 represents random port")
+		rootFlags.String("jaeger", "", "jaeger endpoint url, like: http://localhost:14250/api/traces")
+		if useConfigFile {
```
Comment (gaius-qi) [inline cmd/common/common.go:77]: Why use `useConfigFile` ?

File: cmd/common/common.go [hunk 5]
```diff
@@ -17,88 +17,186 @@
...
+		// Config for binding env
+		viper.SetEnvPrefix(rootName)
+		viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+		_ = viper.BindEnv("config")
 	}
 
 	// Add common cmds
-	cmd.AddCommand(VersionCmd)
-	cmd.AddCommand(newDocCommand(cmd.Name()))
+	if !cmd.HasParent() {
```
Comment (gaius-qi) [inline cmd/common/common.go:93]: Why use `cmd.HasParent`?

File: cmd/common/common.go [hunk 6]
```diff
@@ -17,88 +17,186 @@
...
-		// Use config file from the flag.
-		viper.SetConfigFile(*cfgFile)
-	} else {
-		viper.AddConfigPath(defaultConfigDir)
-		viper.SetConfigName(envPrefix)
-		viper.SetConfigType("yaml")
-	}
+func initConfig(useConfigFile bool, name string, config interface{}) {
+	// Use config file and read once.
+	if useConfigFile && !isConfigFileRead {
```
Comment (gaius-qi) [inline cmd/common/common.go:175]: Why use `useConfigFile` & `isConfigFileRead`?

File: pkg/rpc/dfdaemon/client/client.go [hunk 7]
```diff
@@ -92,7 +92,7 @@ func (dc *daemonClient) getDaemonClientWithTarget(target string) (dfdaemon.Daemo
 func (dc *daemonClient) Download(ctx context.Context, req *dfdaemon.DownRequest, opts ...grpc.CallOption) (*DownResultStream, error) {
 	req.Uuid = uuid.New().String()
 	// 生成taskId
```
Comment (gaius-qi) [inline pkg/rpc/dfdaemon/client/client.go:94]: Use English

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Review by ansinlee — APPROVED** [review]

**Review by 244372610 — APPROVED** [review]

**Review by ansinlee — DISMISSED** [review]

File: cmd/cdn/cmd/root.go (hunk 2 above)
Comment (garfield009) [reply cmd/cdn/cmd/root.go]: final function

**Review by 244372610 — DISMISSED** [review]

**Review by 244372610 — APPROVED** [review]

**Review by ansinlee — APPROVED** [review]

**Pull request merged**